orderer/common/filerepo/filerepo.go

```
@@ -0,0 +1,102 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hyperledger/fabric/common/configtx"
+)
+
+type Repo struct {
+	Options Options
+}
+
+type Options struct {
+	FileSuffix          string
+	TransientFilePrefix string
+	FileRepoDir         string
+}
+
+// Save atomically persists the content to the channelName+suffix file by first writing it
+// to a tmp file prefixed by the TransientFilePrefix and then moves the file to the final
+// destination indicated by the FileSuffix.
+func (r *Repo) Save(channelName string, content []byte) error {
+	err := configtx.ValidateChannelID(channelName)
+	if err != nil {
+		return err
+	}
+
+	fileRepoDir := r.Options.FileRepoDir
+	dest := filepath.Join(fileRepoDir, channelName+r.Options.FileSuffix)
```

I think we want to fail early if `dest` above already exists, otherwise the os.Rename() will fail us later.